openstack/utils/templates/snippets/_proxysql.cfg.tpl

```diff
@@ -81,5 +82,14 @@ mysql_query_rules =
         cache_ttl = 3600000,
         match_pattern = "^SELECT 1$",
     },
+{{- if and .global.Values.proxysql.query_rules .global.Values.proxysql.query_rules.retries_on_select }}
+    {
+        rule_id = 1,
+        active = 1,
+        apply = 1,
+        retries = {{ default 10 .global.Values.proxysql.query_rules.retries_on_select }},
+        match_digest = "^SELECT",
```

Comment:
With [SELECT FOR UPDATE](https://dev.mysql.com/doc/refman/8.4/en/select.html) this can be a DML command and than the retries can be risky if something has been changed in between, or?

Reply:
My assumption was, that retries are only done, if it isn't within a transaction. `SELECT FOR UPDATE` only creates a lock within a transaction.
I've checked the code, and it looks to me, my assumption is [correct](https://github.com/sysown/proxysql/blob/5cc418bb6c59aefca2bc21232e21de6f240a49cf/lib/MySQL_Session.cpp#L4667C56-L4667C75).

A retry of a single statement within an aborted transaction wouldn't make sense.

Reply:
Yes, looks like it might retry such SELECT FOR UPDATE requests, if they weren't running in the transaction, see  https://github.com/sysown/proxysql/blob/v2.5.5-branch/lib/MySQL_Session.cpp#L4234